### PR TITLE
Seperate benchmarking

### PR DESCRIPTION
**Pull request by nick-pestell**
This PR extends validator.py to enable benchmarking from previously sequenced samples through the CLI.

It also adds a CLI for sequencing only.

### CODE REVIEW

File: sim/validator.py [hunk 1]
```diff
@@ -116,6 +116,34 @@ def pipeline(
...
+    # Validate input
+    if not os.path.exists(results_path):
+        raise Exception("Results path does not exist: ", results_path)
+    if not os.path.exists(genomes_path):
+        raise Exception("Genomes path does not exist: ", genomes_path)
+    if not os.path.exists(output_path):
+        raise Exception("Output path does not exist: ", output_path)
+    genomes = genome.from_directory(genomes_path)
+    # TODO: handle when glob does not return a unique path
+    path = glob.glob(results_path + '/Results_*')[0] + '/'
```
Comment (afishman) [inline sim/validator.py]: This is no longer required. It's handled in `sequenced.from_results_dir(path)`

File: sim/validator.py [hunk 2]
```diff
@@ -116,6 +116,34 @@ def pipeline(
         shutil.rmtree(btb_seq_backup_path)
         shutil.rmtree(results_path)
 
+def benchmark(genomes_path, results_path, output_path):
+    # Validate input
+    if not os.path.exists(results_path):
+        raise Exception("Results path does not exist: ", results_path)
+    if not os.path.exists(genomes_path):
```
Comment (afishman) [inline sim/validator.py]: wondering if this `genome.from_directory(` would be a better place for this?

File: sim/validator.py [hunk 3]
```diff
@@ -116,6 +116,34 @@ def pipeline(
         shutil.rmtree(btb_seq_backup_path)
         shutil.rmtree(results_path)
 
+def benchmark(genomes_path, results_path, output_path):
+    # Validate input
+    if not os.path.exists(results_path):
```
Comment (afishman) [inline sim/validator.py]: same as below, wondering if it would be better to have this in `sequenced.from_directory`?

File: sim/validator.py [hunk 4]
```diff
@@ -126,24 +154,15 @@ def sequence_and_benchmark(btb_seq_path, genomes_path, reads_path, output_path,
...
+    sequence(btb_seq_backup_path, reads_path, results_path)
 
-    # Save
-    # TODO: consolidate with pipeline()
-    stats.to_csv(output_path + '/stats.csv')
-
-    for name, df in site_stats.items():
-        df.to_csv(stats_path + f'/{name}_stats.csv')
+    # Benchmark
+    benchmark(genomes_path, results_path, output_path)
```
Comment (afishman) [inline sim/validator.py:156]: nice. I like this.

**Review by afishman — CHANGES_REQUESTED** [review]
looking great. A couple of small things.

File: sim/validator.py (hunk 1 above)
Comment (nick-pestell) [reply sim/validator.py]: As far as i can tell we still need to append `'/Results_*'` to `results_path`.

Do you just mean validating that `path` contains the btb-seq output?

Comment (afishman) [reply sim/validator.py]: Are you sure? Look at the last couple lines of the sequence method. It runs the glob and then constructs the objects

File: sim/validator.py (hunk 2 above)
Comment (nick-pestell) [reply sim/validator.py]: agree, done

File: sim/validator.py (hunk 3 above)
Comment (nick-pestell) [reply sim/validator.py]: agree done

**Review by afishman — APPROVED** [review]
Lovely Job!

**Pull request merged**